lib/utils.js

```diff
@@ -605,6 +607,111 @@ function decorateWithReadConcern(command, coll, options) {
   }
 }
 
+/*
+function parseMsg(fName, msg, param = undefined) {
+  if (!msg) {
+    return;
+  }
+  let parsed = '';
+  let fillingPlaceholder = false;
+  for (let i = 0; i < msg.length; i++) {
+    const char = msg[i];
+    if (fillingPlaceholder) {
+      fillingPlaceholder = false;
+      switch (char) {
+        case 'f':
+          parsed += fName;
+          break;
+        case 'p':
+          parsed += param;
+          break;
+        default:
+          break;
+      }
+    } else {
+      if (char === '%') {
+        fillingPlaceholder = true;
+      } else {
+        parsed += char;
+      }
+    }
+  }
+  console.log(parsed);
+  return parsed;
+}
+*/
+
+function emitDeprecationWarning(msg) {
+  if (process.emitWarning) {
+    process.emitWarning(msg, 'DeprecationWarning');
+  } else {
+    console.warn(msg);
+  }
+}
+
+function messageHandler(fName, msg, param) {
+  if (msg) {
```

Comment:
Instead of having a user-defined `msg`, lets have a user-defined `messageHandler`